### ISSUE TITLE
Closure object structured binding: data member names are missing

### ISSUE THREAD

**Issue by languagelawyer** [andreasfertig/cppinsights#181]
Let's look a the following code
```c++
int main()
{
	int x = 1, y = 2;
	auto f = [&](auto self) {
		(void)x; (void)y; // make 'em used!
		auto& [a, b] = *self; // a == 1, b == 2. Depends on the previous line.
		return b - a;
	};
	return f(&f); // returns 1
}
```
it is transformed into
```c++
int main()
{
  int x = 1;
  int y = 2;
    
  class __lambda_4_11
  {
    int & x;
    int & y;
    public: inline /*constexpr */ int operator()(__lambda_4_11 * self) const
    {
      static_cast<void>(x);
      static_cast<void>(y);
      __lambda_4_11 & __self6 = *self;
      int && a = __self6.;
      int && b = __self6.;
      return b - a;
    }
    public: __lambda_4_11(int & _x, int & _y)
    : x{_x}
, y{_y}
    {}
    
  };
  
  __lambda_4_11 f = __lambda_4_11{x, y};
  return f.operator()(&f);
}
```
https://cppinsights.io/lnk?code=aW50IG1haW4oKQp7CglpbnQgeCA9IDEsIHkgPSAyOwoJYXV0byBmID0gWyZdKGF1dG8gc2VsZikgewoJCSh2b2lkKXg7ICh2b2lkKXk7IC8vIG1ha2UgJ2VtIHVzZWQhCgkJYXV0byYgW2EsIGJdID0gKnNlbGY7IC8vIGEgPT0gMSwgYiA9PSAyLiBEZXBlbmRzIG9uIHRoZSBwcmV2aW91cyBsaW5lLgoJCXJldHVybiBiIC0gYTsKCX07CglyZXR1cm4gZigmZik7IC8vIHJldHVybnMgMQp9&insightsOptions=&std=undefined&rev=1.0

In the lines 15 and 16:
```c++
      int && a = __self6.;
      int && b = __self6.;
```
data member names are missing. Also, I believe, the types of `a` and `b` should be `int&`, not `int&&`:
```c++
      int & a = __self6.x;
      int & b = __self6.y;
```

Not fixing this is also an option, because it is believed that closure object decomposition should be banned by the Standard.

**Comment by andreasfertig** [comment]
Hello @languagelawyer,

thank you. That is once again an interesting piece of code. I gave it a look and had to scratch my head multiple times to see through it. The sad part is, currently I have no glue how to fix it. I'm willing to choose the "not fixing" card but I will think more about it.

Andreas

**Comment by languagelawyer** [comment]
Here is a relevant clang bugreport https://bugs.llvm.org/show_bug.cgi?id=41765

**Comment by andreasfertig** [comment]
Hello @languagelawyer,

sorry that it took so long. I starred at this issue multiple times. Until yesterday the conclusion was always "I don't know how to fix this". Then yesterday, I revisited it from a different angle and there it is :-) I overlooked the fact, that this is the first time a lambda is decomposed. I knew it, but didn't take it into account. Lambda `FieldDecl` does not have a name in Clang. But in other places C++ Insights already makes them visible. So, the fix in the end turn out to be easy. During this, I also corrected the "&&" into a "&" which it really is.

Thanks for reporting this interesting case!

Andreas
